### PR TITLE
GEODE-9808: Throw appropriate exception in AutoConnectionSourceImpl

### PR DESCRIPTION
**Pull request by DonalEvans**
### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### CODE REVIEW

File: geode-core/src/integrationTest/java/org/apache/geode/cache/client/internal/AutoConnectionSourceImplJUnitTest.java [hunk 1]
```diff
@@ -313,7 +296,7 @@ public void test_DiscoverLocators_whenOneLocatorWasShutdown() throws Exception {
       ArrayList<ServerLocation> locators = new ArrayList<>();
       locators.add(new ServerLocation(InetAddress.getLocalHost().getHostName(), secondPort));
       handler.nextLocatorListResponse = new LocatorListResponse(locators, false);
-      Thread.sleep(500);
+      Thread.sleep(1500);
```
Comment (kirklund) [inline geode-core/src/integrationTest/java/org/apache/geode/cache/client/internal/AutoConnectionSourceImplJUnitTest.java:299]: I think we should try to replace this sleep with a use of await. What background activity is happening during the call to sleep?

Comment (DonalEvans) [reply geode-core/src/integrationTest/java/org/apache/geode/cache/client/internal/AutoConnectionSourceImplJUnitTest.java:299]: Good call. I figured out what to await for here to allow the test to pass.

**Review by nabarunnag — APPROVED** [review]
Approved. 
Just for my edification, will there be a similar issue in run2 in UpdateLocatorListTask, where it calls updateLocatorList with the response but we never check the hasResults field and it simply starts accessing the members and calling methods on it.

**Comment by DonalEvans** [comment]
> Approved. Just for my edification, will there be a similar issue in run2 in UpdateLocatorListTask, where it calls updateLocatorList with the response but we never check the hasResults field and it simply starts accessing the members and calling methods on it.

I don't believe so. If we get a response with no results, then the consequence will be that we'll just mark all our old known locators as bad and not add any new ones (because the response doesn't contain any). This behaviour is unaffected by the changes in this PR, and hasn't caused problems up till now, so I have to assume that it's okay.

**Review by Bill — APPROVED** [review]
Approved

**Pull request merged**